src/Contrl/AccSimulator.f90

```diff
@@ -1468,8 +1497,14 @@ subroutine run_AccSimulator()
 
 !          print*,"gamz: ",gammaz(1)
           !get the distance of the center of all effective bunches/bins
-          distance = zcent*Scxlt
-          dzz = sqrt(1.0d0-1.0d0/gammazavg**2)*Clight*dtless*Dt
+          if(FlagEnergyRange == 1) then
+            call globalrange(frange, fgrange, fgammaz, fzcent, fnp, ibunch)
```

Comment:
Main values `grange, gammazavg, zcent` are not modified, here we use `globalrange()` to calculate new filtered values just for the purpose of setting `distance` and `zzz`